File: SymptoMedic.WebApi/SymptoMedic.DataAccess/Entities/Doctor.cs
﻿using System;
using System.Collections.Generic;

#nullable disable

namespace SymptoMedic.DataAccess.Entities
{
    public partial class Doctor
    {
        public Doctor()
        {
            Appointments = new HashSet<Appointment>();
            DoctorSymptoms = new HashSet<DoctorSymptom>();
            Schedules = new HashSet<Schedule>();
        }

        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string License { get; set; }
        public string PracticeName { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        public string PhoneNumber { get; set; }
        public string DoctorSpeciality { get; set; }
        public string PracticeAddress { get; set; }
        public string PracticeCity { get; set; }
        public string PracticeState { get; set; }
        public int? PracticeZipcode { get; set; }
        public string Certifications { get; set; }
        public string Education { get; set; }
        public string Gender { get; set; }

        public virtual ICollection<Appointment> Appointments { get; set; }
        public virtual ICollection<DoctorSymptom> DoctorSymptoms { get; set; }
        public virtual ICollection<Schedule> Schedules { get; set; }
    }
}

File: SymptomaticApp-UI/src/app/interfaces/doctor.ts
import { Appointment } from "./appointments";
import { Symptom } from "./symptom";

export interface Doctor
{
  id: number,
  firstName: string,
  lastName: string,
  license: string,
  practiceName: string,
  email: string,
  password: string,
  phoneNumber: string,
  doctorSpecialty: string,
  practiceAddress: string,
  practiceCity: string,
  practiceState: string,
  practiceZipcode: number,
  certifications: string,
  education: string,
  gender: string,
  appointments: Appointment[],
  doctorSymptoms: Symptom[]

}
